core/src/com/unciv/logic/map/mapgenerator/NaturalWonderGenerator.kt

```diff
@@ -22,410 +24,199 @@ class NaturalWonderGenerator(val ruleset: Ruleset, val randomness: MapGeneration
+        if (groupUnique == null) {
+            minGroupSize = 1
+            maxGroupSize = 1
+        } else {
+            minGroupSize = groupUnique.getIntParam(0)
+            maxGroupSize = groupUnique.getIntParam(1)
         }
+        val targetGroupSize = if (minGroupSize == maxGroupSize) maxGroupSize
+            else (minGroupSize..maxGroupSize).random(randomness.RNG)
 
-        if (MapGenerator.consoleOutput)
-            println("No suitable location for ${wonder.name}")
-        return null
-    }
+        var convertNeighborsTo: String? = null
+        var convertNeighborsExcept: String? = null
+        var convertUnique = wonder.getMatchingUniques(UniqueType.NaturalWonderConvertNeighbors).firstOrNull()
+        convertNeighborsTo = convertUnique?.getStringParam(0)
+        if (convertNeighborsTo == null) {
+            convertUnique = wonder.getMatchingUniques(UniqueType.NaturalWonderConvertNeighborsExcept).firstOrNull()
+            convertNeighborsExcept = convertUnique?.getStringParam(0)
+            convertNeighborsTo = convertUnique?.getStringParam(1)
+        }
+        if (ruleset.terrains.values.none { it.name == convertNeighborsTo && it.type.isBaseTerrain })
+            convertNeighborsTo = null  // ignore if mod badly defined
 
+        if (suitableLocations.size >= minGroupSize) {
+            val location = suitableLocations.random(randomness.RNG)
+            val list = mutableListOf(location)
+            while (list.size < targetGroupSize) {
+                val allNeighbors = list.flatMap { it.neighbors }.minus(list).toHashSet()
+                val candidates = suitableLocations.filter { it in allNeighbors }
+                if (candidates.isEmpty()) break
+                list.add(candidates.random(randomness.RNG))
+            }
+            if (list.size >= minGroupSize) {
+                list.forEach {
+                    clearTile(it)
+                    it.naturalWonder = wonder.name
+                    it.baseTerrain = wonder.turnsInto!!
+                }
+                if (convertNeighborsTo != null) {
+                    for (tile in location.neighbors) {
+                        if (tile.baseTerrain == convertNeighborsTo) continue
+                        if (tile.baseTerrain == convertNeighborsExcept) continue
+                        if (convertNeighborsTo == Constants.coast)
+                            for (neighbor in tile.neighbors) {
+                                // This is so we don't have this tile turn into Coast, and then it's touching a Lake tile.
+                                // We just turn the lake tiles into this kind of tile.
+                                if (neighbor.baseTerrain == Constants.lakes) {
+                                    neighbor.baseTerrain = tile.baseTerrain
+                                    neighbor.setTerrainTransients()
+                                }
```

Comment:
So if we have a tile turn into coast that touches a lake that touches a lake, the second lake won't become coast? I feel if you do this you either have to convert all lakes (requiring a BFS or something) or no lakes to coast